lib/online_order.rb

```diff
@@ -0,0 +1,85 @@
+require 'csv'
+require_relative 'order'
+require_relative 'customer'
+
+ONLINE_FILE = 'support/online_orders.csv'
+
+module Grocery
+  class OnlineOrder < Order
+    attr_reader :status, :customer
+
+    def initialize(id, products, customer_id, status=:pending)
+      super(id, products)
+      @customer = Customer.find(customer_id)
+      @status = status
+    end
+
+    # if the online order instance has no products, returns the total provided by Order#total (no shipping)
+    # otherwise, returns total provided by Order#total + $10 for shipping
+    def total
+      super
+      if @products.empty?
+        return super
```

Comment:
The call to `super` on line 20 doesn't do anything. You could remove it and have the same results.